apps/condo/domains/news/access/NewsItemScope.js

```diff
@@ -55,51 +52,42 @@ async function canManageNewsItemScopes ({ authentication: { item: user }, origin
-                    deletedAt: null,
-                })
-                if (newsItemIds.length !== newsItems.length) return false
-                organizationIds = uniq(newsItems.map(item => get(item, 'organization')))
+                newsItemIds = originalInput.map(item => get(item, 'data.newsItem.connect.id')).filter(Boolean)
+                if (newsItemIds.length !== originalInput.length) return false
+                newsItemIds = uniq(newsItemIds)
             } else {
                 const newsItemId = get(originalInput, 'newsItem.connect.id')
-                const newsItem = await getByCondition('NewsItem', {
-                    id: newsItemId,
-                    deletedAt: null,
-                })
-                if (!newsItem) return false
-                organizationIds = [get(newsItem, 'organization')]
+                if (!newsItemId) return false
+                newsItemIds = [newsItemId]
             }
+
+            const newsItems = await find('NewsItem', {
+                id_in: newsItemIds,
+                deletedAt: null,
+                organization_is_null: false,
+            })
+            if (newsItemIds.length !== newsItems.length) return false
+            organizationIds = newsItems.map(item => item.organization)
+
         } else if (isSoftDeleteOperation) {
-            if (isBulkRequest) {
-                if (!itemIds || !Array.isArray(itemIds)) return false
-                if (itemIds.length !== uniq(itemIds).length) return false
-
-                const newsItems = await find('NewsItem', {
-                    scopes_some: {
-                        id_in: itemIds,
-                        deletedAt: null,
-                    },
-                    deletedAt: null,
-                })
-                organizationIds = uniq(newsItems.map(newsItem => newsItem.organization))
-            } else {
-                if (!itemId) return false
+            const ids = itemIds || [itemId]
+            if (ids.length !== uniq(ids).length) return false
 
-                const newsItem = await getByCondition('NewsItem', {
-                    scopes_some: {
-                        id: itemId,
-                        deletedAt: null,
-                    },
+            const newsItems = await find('NewsItem', {
+                scopes_some: {
+                    id_in: ids,
                     deletedAt: null,
-                })
-                organizationIds = [newsItem.organization]
-            }
+                },
+                deletedAt: null,
+            })
+
+            if (newsItems.some(item => !item.organization)) return false
+            organizationIds = uniq(newsItems.map(newsItem => newsItem.organization))
         }
 
-        return await checkPermissionsInUserOrganizationsOrRelatedOrganizations(user.id, organizationIds, 'canManageNewsItems')
+        return await checkPermissionsInEmployedOrRelatedOrganizations(context, user, organizationIds, 'canManageNewsItems')
```

Comment:
check that `organizationIds` is not empty

Reply:
This line is already filtering properly
```js
 if (newsItems.some(item => !item.organization)) return false
```

Reply:
`if ([].some(item => !item.organization)) return false` will be skip

Reply:
added additional check